iota-client/src/error.rs

```diff
@@ -49,7 +49,7 @@ impl fmt::Display for Error {
             Error::MissingParameter(s) => write!(f, "Must provide required parameter:{}", s),
             Error::InvalidParameter(s) => write!(f, "Parameter is invalid:{}", s),
             Error::SpentAddress => "Found Spent Address that still has balance.".fmt(f),
-            Error::NodePoolEmpty => "No node available".fmt(f),
+            Error::SyncedNodePoolEmpty => "No node available".fmt(f),
```

Comment:
https://github.com/dtolnay/thiserror is nice to implement Display on errors :)

Reply:
yeah I'm planning on making this change soon, just need to find some time to improve the overall quality of this crate.